zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -1,32 +1,108 @@
+//! Prints Zebra checkpoints as "height hash" output lines.
+//!
+//! Get all the blocks up to network current tip and print the ones that are
+//! checkpoints according to rules.
+//!
+//! Usage: zebra-checkpoints [network] --cli <cli-path>
+//! Where network can be `--regtest`, `mainnet` or `--testnet`.
+//! `--cli` is the path of the zcash-cli binary as a string.
+//! If no network is specified program will use the mainnet.
+//!
+//! zebra-consensus accepts an ordered list of checkpoints, starting with the
+//! genesis block. Checkpoint heights can be chosen arbitrarily.
+
 #![allow(clippy::try_err)]
-use color_eyre::eyre::{eyre, Result};
+use color_eyre::eyre::Result;
+use serde_json::Value;
+use std::process::Stdio;
 use structopt::StructOpt;
 
 mod args;
 
+/// We limit the maximum number of blocks in each checkpoint. Each block uses a
+/// constant amount of memory for the supporting data structures and futures.
+const MAX_CHECKPOINT_HEIGHT_GAP: usize = zebrad::commands::LOOKAHEAD_LIMIT / 2;
+
+/// We limit the memory usage for each checkpoint, based on the cumulative size of
+/// the serialized blocks in the chain. Deserialized blocks are larger, because
+/// they contain pointers and non-compact integers. But they should be within a
+/// constant factor of the serialized size.
+const MAX_CHECKPOINT_BYTE_COUNT: i64 = 256 * 1024 * 1024;
+
+/// Checkpoints must be on the main chain, so we skip blocks that are within the
+/// zcashd reorg limit.
+const BLOCK_REORG_LIMIT: i64 = 100;
+
+// Add network argument if needed
+fn network(mut cmd: std::process::Command, args: &args::Args) -> std::process::Command {
+    if args.testnet {
+        cmd.arg("-testnet");
+    } else if args.regtest {
+        cmd.arg("-regtest");
+    }
+    cmd
+}
+
 fn main() -> Result<()> {
     // todo add tracing setup
```

Comment:
We should probably do this as part of this PR, even if we only setup a Registry + Error Layer. https://github.com/ZcashFoundation/zebra/pull/690/files#diff-0f952e6068cc909f968e4d2115f78032R218-R222 should be all you need to do.

Reply:
Thanks, i added tracing at https://github.com/ZcashFoundation/zebra/pull/740/commits/399b6e41f7ab4fff774f6adc3cf4752696018ede